src/segmentmytif/main.py

```diff
@@ -1,33 +1,88 @@
 import argparse
 from pathlib import Path
+from typing import Literal
 
 import numpy as np
 import dask.array as da
 from numpy import ndarray
 from sklearn.ensemble import RandomForestClassifier
+import geopandas as gpd
+import rioxarray
+import dask
+import xarray as xr
 
 from segmentmytif.features import get_features, FeatureType
 from segmentmytif.logging_config import setup_logger, log_duration, log_array
 from segmentmytif.utils.io import read_geotiff, save_tiff
+from segmentmytif.utils.geospatial import get_label_array
 
 logger = setup_logger(__name__)
 
 
-def read_input_and_labels_and_save_predictions(input_path: Path, labels_path: Path, output_path: Path,
-                                               feature_type=FeatureType.FLAIR, features_path: Path = None,
-                                               **extractor_kwargs) -> None:
+def read_input_and_labels_and_save_predictions(
+    raster_path: Path,
+    pos_labels_path: Path,
+    neg_labels_path: Path,
+    output_path: Path,
+    feature_type=FeatureType.IDENTITY,
+    features_path: Path = None,
+    mode: Literal["normal", "parallel", "safe"] = "normal",
+    chunks: dict = None,
+    chunk_overlap: int = 16,
+    **extractor_kwargs,
+) -> None:
     logger.info("read_input_and_labels_and_save_predictions called with the following arguments:")
     for k, v in locals().items():
         logger.info(f"{k}: {v}")
 
-    input_data, profile = read_geotiff(input_path)
-
-    features = get_features(input_data, input_path, feature_type, features_path, profile, **extractor_kwargs)
-
-    labels, _ = read_geotiff(labels_path)
-    prediction_map = make_predictions(features, labels)
-
-    save_tiff(prediction_map, output_path, profile)
+    if chunks is None:
+        chunks = {"band": 1, "x": 1000, "y": 1000}
+
+    match mode:
+        case "normal":
+            raster = rioxarray.open_rasterio(raster_path)  # Load in memory
+        case "parallel":
+            raster = rioxarray.open_rasterio(raster_path, chunks=chunks)
+        case "safe":
+            dask.config.set(scheduler="synchronous")
+            raster = rioxarray.open_rasterio(raster_path, chunks=chunks)
+        case _:
+            msg = f"Invalid mode: {mode}"
+            raise ValueError(msg)
```

Comment:
I think I would prefer this part to be in a separate function (called `set_compute_mode` for instance), so the `read_input_and_labels_and_save_predictions` is bit easier to read.

Reply:
Now there is a `_set_compute_mode` function, which applies global setup for dask schedular, and then returns the keyword args for `rioxarray.open_rasterio`.